### PR TITLE
Add code action to generate RelaxNG grammar

### LINKED ISSUE #1405: CodeAction to generate RelaxNG RNG file
When XML si not associate to a grammar we have those code actions to generate XSD or DTD

image

It should be nice to have a CodeAction action which generates an rng file (RelaxNG XML syntax)

### PR DESCRIPTION
**Pull request by JessicaJHee**
Peek 2022-12-19 16-07

Fixes #1405 

Signed-off-by: Jessica He <jhe@redhat.com>

### CODE REVIEW

**Comment by JessicaJHee** [comment]
Currently I have the start of the generated file as: 

```
<?xml version="1.0" encoding="UTF-8"?>
<element name="rootElementName" xmlns="http://relaxng.org/ns/structure/1.0">
...
```

I see in other examples they have something like :
```
<?xml version="1.0" encoding="UTF-8"?>
<grammar xmlns="http://relaxng.org/ns/structure/1.0" 
         xmlns:sch="http://purl.oclc.org/dsdl/schematron"
         datatypeLibrary="http://www.w3.org/2001/XMLSchema-datatypes">
    <start>
        <element name="rootElementName">
        ...
```

I wonder if the second case is better/more accepted?

**Comment by angelozerr** [comment]
It looks promising !

I think you should generate grammar element because

* the most rng file that I have seen uses grammar
* we have a snippet in  rng file which générales grammar. To be consistent with this snippet I think it is better to generate grammar element

In your demo you generate an rng prefix. I think yiu should generate without prefix. Please see the rng snippet

File: org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/generators/xml2relaxng/XML2RelaxNGGenerator.java [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+
+	@Override
+	protected void generate(Grammar grammar, RelaxNGGeneratorSettings settings, XMLBuilder schema,
+			CancelChecker cancelChecker) {
+		String prefix = "rng";
+		// <?xml version="1.0" encoding="UTF-8"?>
+		schema.startPrologOrPI("xml");
+		schema.addContent(" version=\"1.0\" encoding=\"UTF-8\"");
+		schema.endPrologOrPI();
+		// <element name="rootElementName" xmlns:rng="http://relaxng.org/ns/structure/1.0">
```
Comment (angelozerr) [inline org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/generators/xml2relaxng/XML2RelaxNGGenerator.java:76]: Please generate the same content than snippet

* without rng prefix
* with grammar element
* with define pattern

**Comment by angelozerr** [comment]
@JessicaJHee could you write the same test than https://github.com/eclipse/lemminx/tree/main/org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/extensions/generators/xml2dtd, to be consistent with XSD and DTD tests generator

**Comment by angelozerr** [comment]
The PR looks very good. If CI build works I m OK to merge the PR.

We will improve it and fix bugs on another PR

**Pull request merged**